### ISSUE TITLE
🚨 📢 Google home endpoints returning 403

### ISSUE THREAD

**Issue by ludeeus** [ludeeus/googledevices#44]
Google has pushed an update to google home devices.

All API endpoints with the exception of `/setup/eureka_info` not returns `403` (Unauthorized).
This change renders this library 95% useless and if not a workaround is discovered this will be deprecated and archived.

**Comment by ludeeus** [comment]
It looks like this issue started to roll out 28.06.19 around 02:24 AM (CET)
https://github.com/home-assistant/home-assistant/issues/24815

**Comment by davericher** [comment]
Does this effect the casting functionality or is the 95% more around the alarms/timers/devices tracking

**Comment by ludeeus** [comment]
Casting?
How does this library handle casting?

**Comment by davericher** [comment]
I have no idea, that is why I am asking, it was a very simple yes/no :D

**Comment by ludeeus** [comment]
Okay, then no.

Unless they also changed that, but that would be outside of this.

**Comment by rithvikvibhu** [comment]
Hey @ludeeus, just wanted to let you know:
https://github.com/rithvikvibhu/GHLocalApi/issues/39#issuecomment-511214195

It's just adding another header and changing the base url. I think Home Assistant allows modules to store settings (like a token)?

**Comment by ludeeus** [comment]
Thanks for that @rithvikvibhu :)

Storing tokens in HA is not a problem.

Getting the token, on the other hand, looks like it may be too much for most users (require a rooted android device)

Currently I do not own any android devices, so I'm not able to test/dig myself.
I'll see if I can load up a VM or something to test this out :)

**Comment by rithvikvibhu** [comment]
Yeah, I'm really interested in how the app first gets the token. Maybe while signing in? Might be able to make the same requests.

**Comment by ludeeus** [comment]
I _think_ my devices are in a limbo state, where both 8008 and 8443 are working (without auth token).

![image](https://user-images.githubusercontent.com/15093472/61217737-a4efb900-a710-11e9-8e15-23aab7d8bb17.png)

Which from what I can tell is super weird, I'll throw up a VM when it stops working for me to investigate further.

**Comment by rithvikvibhu** [comment]
Haha. That's definitely weird. Enjoy it while it lasts xD.
If you don't mind losing that "limbo state", restarting has changed for some people.

Edit: Just checked and even mine has the same behavior. Why can't they just stay on something for a while :/

**Comment by Leatherface75** [comment]
Have you found any solution yet?

**Comment by ludeeus** [comment]
No, not yet, and I doubt I will anytime soon (Unless google decides to give you the option to get this token from their dev pages.)

**Comment by Leatherface75** [comment]
Google Assistant already does that and works in Home Assistant so isn't it possible to get that information with that integration?
Both uses this API what i can see https://developers.google.com/actions/smarthome/reference/rest/

**Comment by ludeeus** [comment]
I don't see a token in any of the responses there.

**Comment by Leatherface75** [comment]
Ok i am no expert but there is a lot of commands you can use there and if there is too hard to get local api working maybe it's possible to get informations over that API instead.
Atleast better than not working at all.

https://developers.google.com/actions/smarthome/reference/rpc/google.home.graph.v1

What i think it does is running something like this when starting the google home app or similar command to get information of all devices.

https://developers.google.com/actions/smarthome/reference/rest/v1/devices/sync

**Comment by Leatherface75** [comment]
Here i found something too

https://developers.google.com/actions/smarthome/concepts/local

**Comment by tyjtyj** [comment]
Since GoogleLocalApi is now working 
https://github.com/rithvikvibhu/GHLocalApi

Anyway to bring this integration working ?

**Comment by eliseomartelli** [comment]
That's a great news! @tyjtyj 
I don't have a Google Home on hand now so I can't test it. 
But if I've some time I'd love to help!

**Comment by ludeeus** [comment]
I have a few around @eliseomartelli I'll see if I can do some restructuring here this week to change the port and add an authorization header to the requests.

**Comment by ludeeus** [comment]
Archiving the project.
